lib/p2p/Pool.ts

```diff
@@ -72,6 +72,8 @@ class Pool extends EventEmitter {
   public handshakeData!: HandshakeState;
   /** A map of pub keys to nodes for which we have pending outgoing connections. */
   private pendingOutgoingConnections = new Map<string, Peer>();
+  /** A set of peers for which we have pending incoming connections. */
+  private pendingIncomingConnections = new Set<Peer>();
```

Comment:
In this case the connection isn't pending, because it was already established.

To make this more clear, maybe we should rename these 2 to `pendingOutboundPeers` / `pendingInboundPeers`.